AnkiDroid/src/main/java/com/ichi2/libanki/Note.kt

```diff
@@ -302,18 +303,20 @@ class Note : Cloneable {
         val csum = csumAndStrippedFieldField.second
         // find any matching csums and compare
         val strippedFirstField = csumAndStrippedFieldField.first
-        val fields = col.db.queryStringList(
-            "SELECT flds FROM notes WHERE csum = ? AND id != ? AND mid = ?",
-            csum,
-            this.id,
-            mid
-        )
-        for (flds in fields) {
-            if (Utils.stripHTMLMedia(
-                    Utils.splitFields(flds)[0]
-                ) == strippedFirstField
-            ) {
-                return DupeOrEmpty.DUPE
+        withCol {
+            val fields = col.db.queryStringList(
```

Comment:
I fear I was not clear. You should not use `col`. Instead, `withCol` provides `col` as the receiver of methods. So `db` is implicitly interpreted as `col.db` for example.

It probably does not matter, as, most of the time, `col` and the collection provided by `withCol` are the same collection. But if the collection was closed for whatever reason, both objects can differ; in which case you're using a value that is not valid anymore

Reply:
```suggestion
            val fields = this.db.queryStringList(
```
Can I use like this? Here this belongs to `withCol`

Reply:
The receiver of `withCol` is a `Collection`.

@Arthur-Milchior defined `collection.col` in e90c2dd64066325c52fa13154a9c934927c49fd7, which means that `withCol { col.db }` works using the receiver's `col` rather than the Activity's.

(@Arthur-Milchior, I fear I don't understand the issue here, given the definition of `collection.col`): did we standardise on not using `col` when performing to coroutine migration?

https://github.com/ankidroid/Anki-Android/blob/82951b853bb04329ffac2e2416cd9ba53732aa69/AnkiDroid/src/main/java/com/ichi2/libanki/Collection.kt#L2524-L2529

Reply:
My bad, I totally forgot it. So I guess `col` was right. Seems not intuitive in the first place; but I guess ideally, `Card` and `Note` should not keep `col` anyway, and once it's resolve, there won't be ambiguity